internal/exec/stages/disks/disks.go

```diff
@@ -222,6 +231,82 @@ func (s stage) createRaids(config types.Config) error {
 	return nil
 }
 
+// createLukss creates any LUKS partitions described in config.Storage.Luks
+func (s stage) createLukss(config types.Config) error {
+	lukss := make([]types.Luks, 0, len(config.Storage.Luks))
+	for _, luks := range config.Storage.Luks {
+		if len(luks.Device) > 0 {
```

Comment:
first, this is cool, adding luks support is great.

I was just drive-by glancing at this and you seem to be redundantly validating the device length here, why?  let the validation worry about that, you should probably just use the device paths.

and move the `if len(lukss) == 0 { return nil }` from down below to up top and instead checking `len(config.Storage.Luks)`.